src/compiler/scala/tools/nsc/backend/jvm/BCodeBodyBuilder.scala

```diff
@@ -530,6 +530,7 @@ abstract class BCodeBodyBuilder extends BCodeSkelBuilder {
               if (cast) {
                 mnode.visitTypeInsn(asm.Opcodes.NEW, jlClassCastExceptionRef.internalName)
                 bc dup ObjectRef
+                mnode.visitMethodInsn(asm.Opcodes.INVOKESPECIAL, jlClassCastExceptionRef.internalName, INSTANCE_CONSTRUCTOR_NAME, "()V", true)
```

Comment:
This certainly looks right, but I wonder how we can end up here. We'd need an `x.asInstanceOf[T]` where `x` has primitive type. Any idea? When I write `1.asInstanceOf[String]`, the value is boxed.